### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by NasyTheSlayer**
[DEMO LINK](https://NasyTheSlayer.github.io/react_dynamic-list-of-todos/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,14 +1,52 @@
...
+    getTodos()
+      .then(data => setTodos(data))
+      .finally(() => setLoading(false));
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    let todosCopy = [...todos];
+
+    if (query) {
+      todosCopy = todosCopy.filter((todo) => todo.title.toLowerCase().includes(query.toLowerCase()));
```
Comment (xiimiik) [inline src/App.tsx:32]:
```suggestion
      todosCopy = todosCopy.filter((todo) => (
        todo.title.toLowerCase().includes(query.toLowerCase())
      ));
```

File: src/App.tsx [hunk 2]
```diff
@@ -1,14 +1,52 @@
...
+  const [loading, setLoading] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filter, setFilter] = useState<Status>();
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    setLoading(true);
+    getTodos()
+      .then(data => setTodos(data))
+      .finally(() => setLoading(false));
```
Comment (xiimiik) [inline src/App.tsx:25]: use async await instead

**Review by xiimiik — CHANGES_REQUESTED** [review]
clicking on person name broke a page
Снимок экрана 2023-11-17 в 16 43 52

**Review by nazarmatsevych — APPROVED** [review]
Good work!

**Pull request closed without merging**